### ISSUE TITLE
Not able to save trained model 

### ISSUE THREAD

**Issue by grvsh02** [meta-llama/llama-recipes#752]
### System Info

**System Info:**
GPU used : A100 80 GB
Pytorch version : 2.4.1+cu124
CUDA used to build PyTorch: 12.1

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### 🐛 Describe the bug

After training meta-llama/Llama-3.2-11B-Vision-Instruct using a single GPU fine tuning, the model is not saved in the output dirs

The command used is :

`python -m recipes.quickstart.finetuning.finetuning --lr 1e-5  --num_epochs 3 --batch_size_training 2 --model_name meta-llama/Llama-3.2-11B-Vision-Instruct --use_fast_kernels --dataset "custom_dataset" --custom_dataset.test_split "test" --custom_dataset.file "path/to/dataset/file.py" --run_validation False --batching_strategy padding  --use_peft --peft_method lora --output_dir path/to/model_output
`

After the model finetuning finishes, there is nothing in the output dirs

### Error logs

No errors were logged, the finetuning finished with no errors.

### Expected behavior

An output dir to be created with model weights and config

**Comment by mreso** [comment]
Hi @grvsh02 Can you post the output of your finetuning run?

**Comment by grvsh02** [comment]
Here is the finetune output

`gaurav.sharma@vm-gpu-a100-cen-24v:~/llama-recipes$ python -m recipes.quickstart.finetuning.finetuning --lr 1e-5  --num_epochs 3 --batch_size_training 2 --model_name meta-llama/Llama-3.2-11B-Vision-Instruct --use_fast_kernels --dataset "custom_dataset" --custom_dataset.test_split "test" --custom_dataset.file "/home/gaurav.sharma/docsai-invoice-models/pre_processing/llama_3_2_vision/recipe.py" --run_validation False --batching_strategy padding  --use_peft --peft_method lora --output_dir /home/gaurav.sharma/llama_3_2_vision_fsdp
/home/gaurav.sharma/lib/python3.12/site-packages/llama_recipes/model_checkpointing/checkpoint_handler.py:17: DeprecationWarning: `torch.distributed._shard.checkpoint` will be deprecated, use `torch.distributed.checkpoint` instead
  from torch.distributed._shard.checkpoint import (
Loading checkpoint shards: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 5/5 [00:03<00:00,  1.65it/s]
--> Model meta-llama/Llama-3.2-11B-Vision-Instruct

--> meta-llama/Llama-3.2-11B-Vision-Instruct has 10670.220835 Million params

trainable params: 5,898,240 || all params: 10,676,119,075 || trainable%: 0.0552
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 226/226 [00:27<00:00,  8.22it/s]
!!! Dataset loaded successfully !!!
--> Training Set Length = 226
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 226/226 [00:26<00:00,  8.60it/s]
!!! Dataset loaded successfully !!!
--> Validation Set Length = 226
length of dataset_train 226
custom_data_collator is used
--> Num of Training Set Batches loaded = 113
Starting epoch 0/3
train_config.max_train_step: 0
/home/gaurav.sharma/lib/python3.12/site-packages/torch/cuda/memory.py:365: FutureWarning: torch.cuda.reset_max_memory_allocated now calls torch.cuda.reset_peak_memory_stats, which resets /all/ peak memory stats.
  warnings.warn(
Training Epoch: 1:   0%|                                                                                                                                              | 0/113 [00:00<?, ?it/s]/home/gaurav.sharma/lib/python3.12/site-packages/torch/autograd/graph.py:825: UserWarning: cuDNN SDPA backward got grad_output.strides() != output.strides(), attempting to materialize a grad_output with matching strides... (Triggered internally at ../aten/src/ATen/native/cudnn/MHA.cpp:674.)
  return Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
Training Epoch: 1/3, step 112/113 completed (loss: 0.40972328186035156): 100%|██████████████████████████████████████████████████████████████████████████████| 113/113 [07:17<00:00,  3.87s/it]
Max CUDA memory allocated was 58 GB
Max CUDA memory reserved was 66 GB
Peak active CUDA memory was 58 GB
CUDA Malloc retries : 0
CPU Total Peak Memory consumed during the train (max): 19 GB
Epoch 1: train_perplexity=1.7289, train_epoch_loss=0.5475, epoch time 437.7825730930001s
Starting epoch 1/3
train_config.max_train_step: 0
Training Epoch: 2/3, step 112/113 completed (loss: 0.22428105771541595): 100%|██████████████████████████████████████████████████████████████████████████████| 113/113 [04:06<00:00,  2.18s/it]
Max CUDA memory allocated was 58 GB
Max CUDA memory reserved was 64 GB
Peak active CUDA memory was 58 GB
CUDA Malloc retries : 0
CPU Total Peak Memory consumed during the train (max): 20 GB
Epoch 2: train_perplexity=1.4511, train_epoch_loss=0.3723, epoch time 247.0452979700001s
Starting epoch 2/3
train_config.max_train_step: 0
Training Epoch: 3/3, step 112/113 completed (loss: 0.5427780151367188): 100%|███████████████████████████████████████████████████████████████████████████████| 113/113 [04:06<00:00,  2.18s/it]
Max CUDA memory allocated was 58 GB
Max CUDA memory reserved was 62 GB
Peak active CUDA memory was 58 GB
CUDA Malloc retries : 0
CPU Total Peak Memory consumed during the train (max): 20 GB
Epoch 3: train_perplexity=1.3742, train_epoch_loss=0.3179, epoch time 247.16199585799995s
Key: avg_train_prep, Value: 1.5180402199427288
Key: avg_train_loss, Value: 0.4125426709651947
Key: avg_epoch_time, Value: 310.66328897366674
Key: avg_checkpoint_time, Value: 0`

**Comment by mreso** [comment]
Thanks for the logs. Which version are you running? We recently switched the saving behavior to save the model even if we do not run validation. If you're running 0.4.0.post1 or prior you do not have that change yet and you will need to run validation to save the model. Alternatively you can update llama-recipes to the current tip of the main branch:
```
pip install -U git+https://github.com/meta-llama/llama-recipes
```
That will save the model with your current command line without running validation.

**Comment by grvsh02** [comment]
I figured out that it was due to the validation check being False. Therefore enabled it and got it working. Thanks !
